src/ProfiledAot/src/maui/MainPage.xaml.cs

```diff
@@ -3,16 +3,39 @@ namespace maui;
 public partial class MainPage : ContentPage
 {
 	int count = 0;
+	bool _isLoaded;
+	bool _startCompleted;
 
 	public MainPage()
 	{
 		InitializeComponent();
-		Start();
+
+		if (Application.Current.MainPage is not AppFlyoutPage)
+		{
+			Start();
+			this.Loaded += OnMainPageLoaded;
+		}
 	}
 
 	private async void Start()
 	{
 		CounterBtn.Text = await CommonMethods.Invoke();
+		_startCompleted= true;
+		LoadFlyoutPage();
+	}
+
+	void OnMainPageLoaded(object sender, EventArgs e)
+	{
+		_isLoaded = true;
+		LoadFlyoutPage();
```

Comment:
Should we try a `BeginInvoke` or `Task.Delay(100)` here to see if we get more methods in the profile?

We can also increase this time if needed:

https://github.com/dotnet/maui/blob/83ae546b3add4c81580efdfbd550376eaaafcba0/src/ProfiledAot/src/Directory.Build.targets#L31